src/dishka/integrations/aiogram.py

```diff
@@ -108,7 +108,8 @@ def setup_dishka(
 
     if auto_inject:
         callback = partial(inject_router, router=router)
-        router.startup.register(callback)
+        for sub_router in router.chain_tail:
```

Comment:
Let's move loop to the callback, otherwise we implicitly require `setup_dishka` be the latest call, when all routers are included